### DISCUSSION TITLE
Duplicated dependencies across pages in the server bundle

### DISCUSSION THREAD

**Discussion by dalmo3** [vercel/next.js#18076] [Help] [answered]
First time using Next. When running the bundle analyzer I was surprised to see the amount of duplicated code in the server bundle:

![image](https://user-images.githubusercontent.com/33860060/96662570-2077f580-13ab-11eb-9ad4-553e7d5bf582.png)

That is basically from following the [Learn](https://nextjs.org/learn/basics/create-nextjs-app) tutorial on the website, so the [demo repo](https://github.com/vercel/next-learn-starter/tree/master/demo) can reproduce this as well: 

![image](https://user-images.githubusercontent.com/33860060/96663073-65e8f280-13ac-11eb-8600-e472b447fd67.png)

This is related to both https://github.com/vercel/next.js/issues/5923 and https://github.com/vercel/next.js/issues/4684. I tried setting this with no luck:

```
module.exports = {
  experimental: {
    granularChunks: true
  }
}
```

Then I read through those issues again and it seems they dealt with the client bundle. My client side chunks look fine, though.

So the question is: Is that by design? Are the pages meant to be self-contained for some sort of serverless architecture? If that's the case where can I learn more about it?

**Comment by timneutkens** [comment, marked as answer]
Webpack does not support splitchunks for the Node.js target in webpack 4 hence why bundled code is included in every bundle. Webpack 5 support is currently in beta and we're planning to add splitchunks for server bundles to that.

**Reply by thiagozf** [reply]
@timneutkens Do you have a rough ETA for when this will be available?
